### PR TITLE
Expand add env functionality to add variables if not present

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
currently apply-config-from-env does not provide a way to inject new variables in the config during testing phase. This makes testing different scenarios very cumbersome. This pr adds this capability
### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: docker/pulsar/scripts/apply-config-from-env.py [hunk 1]
```diff
@@ -34,6 +34,9 @@
 # Always apply env config to env scripts as well
 conf_files = ['conf/pulsar_env.sh', 'conf/bkenv.sh'] + sys.argv[1:]
 
+PF_ENV_PREFIX = 'INS_'
```
Comment (merlimat) [inline docker/pulsar/scripts/apply-config-from-env.py]: Can we use the prefix `PULSAR_` ?

Comment (srkukarni) [reply docker/pulsar/scripts/apply-config-from-env.py]: changed

**Comment by jerrypeng** [comment]
@srkukarni seems to be failing org.apache.pulsar.tests.integration.presto.TestBasicPresto for some reason

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at
780ba31

**Pull request merged**